File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/docker/DockerExecutableSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.docker;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableDockerExecutableSettings.class)
public interface DockerExecutableSettings {

  Optional<String> entrypoint();

  Optional<String> command();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/configurations/docker/DockerExecutableSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.docker

import com.fasterxml.jackson.annotation.JsonProperty

data class DockerExecutableSettings(

    @JsonProperty("Entrypoint")
    val entrypoint: String,

    @JsonProperty("Command")
    val command: String
)
